File: chapter5/java/designpattern/src/main/java/com/designpattern/CheesePizza.java
package com.designpattern;

/**
 * CheesePizza is a concrete class extending abstract class Pizza. It implements abstract methods as per the requirement to prepare a Cheese pizza.
 *
 * @Author Bridget Wu
 */
public class CheesePizza extends Pizza {
    /**
     * add Cheese topping
     */
    @Override
    void addTopping() {
        System.out.println("=== Add Cheese topping === ");
    }
}

File: chapter5/typescript/CheesePizza.ts
import { Pizza } from "./Pizza";

/**
 * CheesePizza is a concrete class extending abstract class Pizza. It implements abstract methods as per the requirement to prepare a Cheese pizza.
 *
 * @Author Bridget Wu
 */
 export class CheesePizza extends Pizza {
    /**
     * add Cheese topping
     */
     addTopping(): void {
        console.log("=== Add Cheese topping === ");
    }
}
